R/json_pack.R

```diff
@@ -31,6 +31,117 @@ json_pack.data.frame <- function(.data, ..., .names_sep = NULL) {
 to_packed_json <- function(x) {
   con <- textConnection(NULL, open = "w")
   on.exit(close(con))
-  jsonlite::stream_out(x, con, digits = NA)
+  jsonlite::stream_out(x, con, digits = NA, verbose = FALSE)
   textConnectionValue(con)
 }
+
+#' @export
+json_pack.tbl_lazy <- function(.data, ..., .names_sep = NULL) {
+  dots <- quos(...)
+  if ("" %in% names2(dots)) {
+    abort("All elements of `...` must be named.")
+  }
+
+  col_nms <- colnames(.data)
+  pack_cols <- purrr::map(dots, ~ tidyselect::vars_select(col_nms, !!.x))
+  id_cols <- setdiff(col_nms, unlist(unique(pack_cols)))
+
+  sql_exprs <- purrr::imap(pack_cols, ~ sql_json_pack(
+    dbplyr::remote_con(.data),
+    cols = names(.x),
+    names_sep = .names_sep,
+    packed_col = .y,
+    data = .data
+  ))
+
+  .data %>%
+    transmute(!!!syms(id_cols), !!!sql_exprs)
+}
+
+sql_json_pack <- function(con, cols, names_sep, packed_col, data) {
```

Comment:
All methods need an ellipsis, otherwise it's difficult to extend once they are exported. (We don't need it right now, but it's so easy to forget later.)